flutter_vlc_player/example/lib/single_tab.dart

```diff
@@ -0,0 +1,198 @@
+            print('onRendererHandler $type $id $name');
+          },
+        );
+        break;
+      case VideoType.asset:
+        _controller = VlcPlayerController.asset(
+          initVideo.path,
+          // package: "test",
+          onInit: () async {
+            await _controller.startRendererScanning();
+          },
+          onRendererHandler: (type, id, name) {
+            print('onRendererHandler $type $id $name');
+          },
+          options: VlcPlayerOptions(),
+        );
+        break;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView(
+      children: [
+        Container(
+          height: 400,
+          child: VlcPlayerWithControls(key: _key, controller: _controller),
+        ),
+        ListView.builder(
+          shrinkWrap: true,
+          itemCount: listVideos.length,
+          physics: NeverScrollableScrollPhysics(),
+          itemBuilder: (BuildContext context, int index) {
+            VideoData video = listVideos[index];
+            IconData iconData;
+            switch (video.type) {
+              case VideoType.network:
+                iconData = Icons.cloud;
+                break;
+              case VideoType.file:
+                iconData = Icons.insert_drive_file;
+                break;
+              case VideoType.asset:
+                iconData = Icons.all_inbox;
+                break;
+            }
+            return ListTile(
+              selected: selectedVideoIndex == index,
+              selectedTileColor: Colors.black54,
+              leading: Icon(
+                iconData,
+                color:
+                    selectedVideoIndex == index ? Colors.white : Colors.black,
+              ),
+              title: Text(
+                video.name,
+                overflow: TextOverflow.ellipsis,
+                style: TextStyle(
+                  color:
+                      selectedVideoIndex == index ? Colors.white : Colors.black,
+                ),
+              ),
+              subtitle: Text(
+                video.path,
+                overflow: TextOverflow.ellipsis,
+                style: TextStyle(
+                  color:
+                      selectedVideoIndex == index ? Colors.white : Colors.black,
+                ),
+              ),
+              onTap: () async {
+                switch (video.type) {
+                  case VideoType.network:
+                    await _controller.setMediaFromNetwork(video.path,
+                        hwAcc: HwAcc.FULL);
+                    break;
+                  case VideoType.file:
+                    File file = File(video.path);
+                    if (await file.exists())
+                      await _controller.setMediaFromFile(file);
+                    else
+                      Scaffold.of(context).showSnackBar(
+                        SnackBar(
+                          content: Text("File not exists!"),
```

Comment:
Nit: "File does not exist!"

Looks like the option to play the video from file in the example app is not portable. 
I'm not sure we can ensure that a video is present on the file system when installing the app, but we can do that upon launch.

There are a couple of options here:
1. Include a small video as an asset and extract the video to filesystem upon launch.
2. Use file_picker or image_picker, so that the user could supply a video themselves.

I'm personally in favor of option 1. It will introduce a small delay, since the copying process isn't instant, but it still feels like a 'batteries included' example, compared to option 2, where the user will have to ensure they have a video somewhere on the filesystem.

Reply:
What's the pseudo code for option 1 ?

Reply:
Something like this:
```dart
void initState() {
  _fsVideoFuture = _loadVideoToFs();
}

Future<File> _loadVideoToFs() async {
  final videoData = await rootBundle.load('fs-video');
  final videoBytes = Uint8List.view(videoData.buffer);
  final dir = Directory.systemTemp;
  final videoFile = File(join(dir, 'fs-video.mp4'))
    ..createSync()
    ..writeAsBytesSync(videoBytes);
  return videoFile;
}
```

We can use the Future show a progress indicator while the video is loading.

Reply:
we will try it!

Reply:
Have this mostly working, discussing with @alr2413 on the side before I push up. Will finish up tomorrow.